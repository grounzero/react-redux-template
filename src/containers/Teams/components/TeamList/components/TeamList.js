import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { Table } from 'semantic-ui-react';

const TeamList = (props) => {
  const { teams } = props;
  const { error, loading, models } = teams;

  if (error) {
    return (
      <p>Sorry! There was an error loading the teams</p>
    );
  }

  if (loading) {
    return (
      <div className='ui active inverted dimmer'>
        <div className='ui text loader'>Loading</div>
      </div>
    );
  }

  return (
    <Table compact celled>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>Name</Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {models.map(team => (
          <Table.Row key={team.id}>
            <Table.Cell><NavLink to={'/teams/' + team.id} activeClassName='route--active'>{team.name}</NavLink></Table.Cell>
          </Table.Row>
        ))}
      </Table.Body>
    </Table>
  );
};

TeamList.propTypes = {
  teams: PropTypes.shape({
    models: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired
    })),
    loading: PropTypes.bool.isRequired,
    error: PropTypes.bool.isRequired
  }).isRequired,
  actions: PropTypes.object.isRequired
};


export default TeamList;