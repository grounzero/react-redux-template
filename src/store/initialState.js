const _initialState = {
  global: {
    userInfo: {
      domain: 'express',
      username: 'dev'
    },
    versionInfo: {
      company: 'Sky Graphics & Data Solutions',
      copyright: 'Copyright Â© 2017',
      date: '01 January 2017',
      description: '',
      product: 'Sports All-Stars Stats',
      title: 'Sports All-Stars Stats',
      version: '0.1.0'
    }
  },
  fixtures: {
  },
  teams: {
    models: [],
    loading: false,
    error: false
  }
};

const initialState = process.env.NODE_ENV === 'production' ? window.__INITIAL_STATE__ : _initialState;

export default initialState;