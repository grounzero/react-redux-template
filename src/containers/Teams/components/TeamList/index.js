import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Button, Header, Segment } from 'semantic-ui-react';

import * as thunks from '../../thunks';
import TeamList from './components/TeamList';

import './styles.scss';

class TeamListContainer extends Component {
    componentDidMount() {
        const { teams, actions } = this.props;

        if (teams.models.length === 0) {
            actions.getTeams();
        }
    }

    render() {
        const { teams, actions } = this.props;

        const refresh = () => actions.getTeams(true);
        return (
            <div>
                <Header as='h2' attached='top'>
                    Teams
                    <Button primary onClick={refresh} floated='right'>Refresh</Button>
                </Header>
                <Segment attached>
                    <TeamList teams={teams} actions={actions} />
                </Segment>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        teams: state.teams
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(thunks, dispatch)
    };
};

TeamListContainer.propTypes = {
    teams: PropTypes.shape({
        models: PropTypes.array.isRequired
    }).isRequired,
    actions: PropTypes.object.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(TeamListContainer);