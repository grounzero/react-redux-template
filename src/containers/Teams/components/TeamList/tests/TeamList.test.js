import React from 'react';
import renderer from 'react-test-renderer';

import TeamList from '../components/TeamList';
import initialState from '../../../../../store/initialState';

const { teams } = initialState;
const actions = {};

it('TeamList renders correctly', () => {
    const tree = renderer.create(
        <TeamList teams={teams} actions={actions} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});