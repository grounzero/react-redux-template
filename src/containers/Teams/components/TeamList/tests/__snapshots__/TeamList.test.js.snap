// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamList renders correctly 1`] = `
<table
  className="ui celled compact table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        Name
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  />
</table>
`;
